datadog/dogstatsd/base.py

```diff
@@ -79,6 +100,9 @@ def get_socket(self):
         avoid bad thread race conditions.
         """
         if not self.socket:
+            if self.use_default_route:
+                self.host = self._get_default_route()
+
```

Comment:
This could be a good candidate for a Python *property* too.
```python
    @property.setter
    def use_default_route(self, value):
        self._use_default_route = value
       
       # Invalid the socket 
       self.socket = None

        if self._use_default_route:
            self.host = self._get_default_route()
```
with the following benefits:
1. We don't need to repeat `self.host = self._get_default_route()` twice.
2. The `host` and `socket` attributes would be systematically updated on `use_default_route` changes.